### ISSUE TITLE
NullPointerException in JGitverModelProcessor.provisionModel when POM unavailable.

### ISSUE THREAD

**Issue by JHeiner** [jgitver/jgitver-maven-plugin#30]
## Issue

**version**: 0.3.0-alpha3

**usage context**:

 (everywhere)

**Problem description**:

Sometimes a JAR gets put into a repository without a POM. Our corporate repo holds several third-party JAR libraries that were not created by Maven (or any other tool that knows to generate a POM for it). Maven itself is OK with this. It warns that without a POM the dependency information is incomplete, but happily uses the JAR anyway.

JGitverModelProcessor fails in such cases.

I already have the code to fix this issue. I still need to figure out how to set up a test. So I'll get a pull request in soon as that's done.

**Comment by JHeiner** [comment]
In case I get hit by a meteor, and so you know which direction I'm headed, here's the patch I've applied to `provisionModel`

``` java
        Source source = (Source) options.get(ModelProcessor.SOURCE);
        //logger.debug( "JGitverModelProcessor.provisionModel source="+source );
        if (source == null) {
            return model;
        }

        File location = new File(source.getLocation());
        //logger.debug( "JGitverModelProcessor.provisionModel location="+location );
        if (!location.isFile()) {
            // their JavaDoc says Source.getLocation "could be a local file path, a URI or just an empty string."
            // if it doesn't resolve to a file then calling .getParentFile will throw an exception,
            // but if it doesn't resolve to a file then it isn't under getMultiModuleProjectDirectory,
            return model; // therefore the model shouldn't be modified.
        }

        File relativePath = location.getParentFile().getCanonicalFile();
```

**Comment by McFoggy** [comment]
@JHeiner if you could describe a use case when such a situation can occur, I mean fill a repo with a file without a pom, it would help in reproducing.

**Comment by JHeiner** [comment]
I can't think of a use case that provides motivation for sticking something into a repo without a POM because to me that feels like an error. If I were in charge of Maven I would probably forbid it, but the Maven developers chose to allow it.

Also, I don't know that this questionable but allowed action is the only way to cause the NPE in `provisionModel`... the JavaDoc I referenced above is silent on what conditions cause `getLocation` to return a file path versus a URI. I happened to stumble into one such condition, and that's what I coded into the test.
